src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +687,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #unlock' do
+    before do
+      login user
+    end
+
+    it "project can't be unlocked if it hasn't lock and enable flags" do
+      post :unlock, project: user.home_project
+      is_expected.to redirect_to(action: :show, project: user.home_project)
+      expect(flash[:notice]).to eq("Project can't be unlocked: is not locked")
+    end
+
+    it "maintenance incident can't be unlocked if there are maintenance release requests" do
+      user.home_project.flags.create(flag: 'lock', status: 'enable')
```

Comment:
If you look at the other tests, we also try to make the tests as atomar as possible, e.g.:

``
it { expect(flash[:error]).to eq("Can not remove path: ") }
it { is_expected.to redirect_to(:back) }
``
<- each expect is a separate test which makes debugging a lot easier!